core/src/main/scala/magnolia1/magnolia.scala

```diff
@@ -181,8 +181,24 @@ object Magnolia {
           annotation.tree
       }
 
-    def annotationsOf(symbol: Symbol): List[Tree] =
-      annotationTrees(symbol.annotations)
+    def annotationsOf(symbol: Symbol): List[Tree] = {
+      @tailrec
+      def fromBaseClasses(owner: Symbol): List[Annotation] =
+        if (owner.isClass) {
+          owner.asClass.baseClasses
+            .flatMap(_.asType.toType.members)
+            .collect {
+              case t: TermSymbol if t.name == symbol.name && t.annotations.exists(isInherit) => t.annotations.filterNot(isInherit)
```

Comment:
I wanted to compare `Symbols` with `==` but that didn't work so I compare just by `name` but if feels risky to do just that

Reply:
hm I think if we're looking only at the symbols defined in super-classes / super-traits that's fine. Although, if the symbol is a method, we might also compare the number or aguments or agument types. That will be an exact match. Maybe there's a way to obtain the method's signature and compare that?

Reply:
although, signatures aren't 100% fail-safe either, as an overriding method may specialise parameters. But since we are looking at no-arg defs anyway, probably just comparing the name & arity is good enough :)